presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkStreamReaders.java

```diff
@@ -48,548 +55,666 @@
+                throws Exception
         {
-            OrcDataSource dataSource = new FileOrcDataSource(tinyIntNoNullFile, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), true);
-            OrcReader orcReader = new OrcReader(dataSource, ORC, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE));
-            return orcReader.createRecordReader(
-                    ImmutableMap.of(0, TINYINT),
-                    OrcPredicate.TRUE,
-                    UTC, // arbitrary
-                    newSimpleAggregatedMemoryContext(),
-                    INITIAL_BATCH_SIZE);
+            setup(TINYINT);
         }
 
-        private List<Byte> createTinyIntValuesNoNull()
+        @Override
+        protected Iterator<?> createValues()
         {
             List<Byte> values = new ArrayList<>();
             for (int i = 0; i < ROWS; ++i) {
                 values.add(Long.valueOf(random.nextLong()).byteValue());
             }
-            return values;
+            return values.iterator();
         }
     }
 
     @SuppressWarnings("FieldMayBeFinal")
     @State(Scope.Thread)
     public static class TinyIntWithNullBenchmarkData
+            extends BenchmarkData
     {
-        protected File temporaryDirectory;
-        protected File tinyIntWithNullFile;
-        private Random random;
-
         @Setup
         public void setup()
                 throws Exception
         {
-            random = new Random(0);
-            temporaryDirectory = createTempDir();
-            tinyIntWithNullFile = new File(temporaryDirectory, randomUUID().toString());
-            writeOrcColumnHive(tinyIntWithNullFile, ORC_12, NONE, TINYINT, createTinyIntValuesWithNull().iterator());
-        }
-
-        @TearDown
-        public void tearDown()
-                throws IOException
-        {
-            deleteRecursively(temporaryDirectory.toPath(), ALLOW_INSECURE);
-        }
-
-        private OrcRecordReader createRecordReader()
-                throws IOException
-        {
-            OrcDataSource dataSource = new FileOrcDataSource(tinyIntWithNullFile, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), true);
-            OrcReader orcReader = new OrcReader(dataSource, ORC, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE));
-            return orcReader.createRecordReader(
-                    ImmutableMap.of(0, TINYINT),
-                    OrcPredicate.TRUE,
-                    UTC, // arbitrary
-                    newSimpleAggregatedMemoryContext(),
-                    INITIAL_BATCH_SIZE);
+            setup(TINYINT);
         }
 
-        private List<Byte> createTinyIntValuesWithNull()
+        @Override
+        protected Iterator<?> createValues()
         {
             List<Byte> values = new ArrayList<>();
             for (int i = 0; i < ROWS; ++i) {
```

Comment:
This creates all null values. Original tests were targeted for values with interleaved nulls. The same problem applies to other types too. I think the all null cases are covered by readAllNull and we should have a case that covers partial nulls.